merge.go

```diff
@@ -11,6 +11,7 @@ package mergo
 import (
 	"fmt"
 	"reflect"
+	"unsafe"
```

Comment:
That worries we :sweat_smile:

Reply:
I will look for a better solution, the fields are immutable that is why I did this, and my experience with reflect is still limited.

Reply:
@vdemeester any idea on how to implement this without unsafe? I cannot fin a way to access private struct fields. Without accessing them, we cannot copy the private fields and as a result mergo would not work as expected. If you know some way, or have an alternative solution please share it with me. Thank you!